src/3d/symbols/qgsline3dsymbol_p.cpp

```diff
@@ -150,4 +154,105 @@ Qt3DRender::QGeometryRenderer *QgsLine3DSymbolEntityNode::renderer( const Qgs3DM
   return renderer;
 }
 
+
+Qt3DRender::QGeometryRenderer *QgsLine3DSymbolEntityNode::rendererSimple( const Qgs3DMapSettings &map, const QgsLine3DSymbol &symbol, const QgsVectorLayer *layer, const QgsFeatureRequest &request )
+{
+  QVector<QVector3D> vertices;
+  vertices << QVector3D();  // the first index is invalid, we use it for primitive restart
+  QVector<unsigned int> indexes;
+
+  QgsPoint centroid;
+  QgsPointXY origin( map.origin().x(), map.origin().y() );
+  QgsFeature f;
+  QgsFeatureIterator fi = layer->getFeatures( request );
+  while ( fi.nextFeature( f ) )
+  {
+    if ( f.geometry().isNull() )
+      continue;
+
+    if ( symbol.altitudeBinding() == AltBindCentroid )
+      centroid = QgsPoint( f.geometry().centroid().asPoint() );
+
+    QgsGeometry geom = f.geometry();
+    const QgsAbstractGeometry *g = geom.constGet();
+    if ( QgsLineString *ls = qgsgeometry_cast<QgsLineString *>( g ) )
```

Comment:
shouldn't this be 

    if ( const QgsLineString *ls = qgsgeometry_cast< const QgsLineString *>( g ) )

(Does qgsgeometry_cast allow casting away const? (it's a bug if it does). And if not, this cast would/should fail?)

Reply:
It seems that's by design of qgsgeometry_cast - its definition includes const_cast - which probably should not be there. Changing this to const LineString...

Reply:
@m-kuhn any feedback here? Can we remove the const_cast?

Reply:
Go ahead. I think the compiler will tell you quickly what the reason for introducing this was (if any) and will tell you at the same speed if alternative approaches do work or not.